servlet/src/main/java/io/undertow/servlet/api/DeploymentInfo.java

```diff
@@ -1399,6 +1400,15 @@ public void setPreservePathOnForward(boolean preservePathOnForward) {
         this.preservePathOnForward = preservePathOnForward;
     }
 
+    public int getFileCacheTimeToLive() {
```

Comment:
I think we have two different Jiras in this commit: one of them is adding support to configuring the cache parameters, the other one is fixing the bug

Reply:
@fl4via in the end we need to know the TTL used and apply the same value in the cache inside `ServletPathMatches`. Initially I just tried something like the following:

```java
this.pathMatchCache = new LRUCache<>(1000, this.resourceManager instanceof CachingResourceManager? ((CachingResourceManager) this.resourceManager).getMaxAge() : -1, true);
```

So if it was a caching one I retrieved the TTL from it and if not -1 (cache forever) was used. But I think there are more complicated situations that can use a `DelegatingResourceManager` ([here](https://github.com/wildfly/wildfly/blob/23.0.2.Final/undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java#L568-L577)). And that's way I considered better to add the TTL in the deployment info and be sure we are using it correctly.

Reply:
Finally we are going back to the cast idea to get the TTL from the `CachingResourceManager`. We don't want to change the `DeploymentInfo` API.

Reply:
The changes in teh `DeploymentInfo` have ben removed and teh TTL is obtained from the `resourceManager` if it's a `CachingResourceManager, if not -1 (cache forever) is used as before.

Reply:
Thank you @rmartinc!